pkg/apis/core/v1/resource_tags_test.go

```diff
@@ -0,0 +1,86 @@
+package v1
+
+import (
+	"context"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/mock"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	clouddnsv1 "go.anx.io/go-anxcloud/pkg/apis/clouddns/v1"
+	"go.anx.io/go-anxcloud/pkg/client"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Generic API resource tags", func() {
+	Context("mock", func() {
+		var mockAPIClient mock.MockAPI
+
+		BeforeEach(func() {
+			a, err := mock.NewMockAPI(true)
+			Expect(err).ToNot(HaveOccurred())
+			mockAPIClient = a
+		})
+
+		It("tags, untags and list tags of an object", func() {
+			z := &clouddnsv1.Zone{Name: "abc"}
+
+			err := Untag(context.TODO(), mockAPIClient, z, "tag-1", "tag-2", "tag-3")
+			Expect(err).ToNot(HaveOccurred())
+
+			err = Untag(context.TODO(), mockAPIClient, z, "tag-2", "tag-3")
+			Expect(err).ToNot(HaveOccurred())
+
+			err = Tag(context.TODO(), mockAPIClient, z, "tag-2", "tag-3")
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(mockAPIClient.OperationPathRequestCount(types.OperationCreate, "/api/core/v1/resource.json/abc/tags/tag-1")).To(Equal(0))
```

Comment:
something like `mockAPIClient.Expect(ResourceWithTag{Identifier: "abc", Tag: "tag-1"}).To(BeCreated())` could be great

Reply:
maybe it's more like `mockAPIClient.Expect(&corednsv1.Zone{Name: "example.com", AdminEmail: "mara@example.com"}).To(BeCreated("Name", "AdminEmail"))`, setting which fields have to match

important: you may have to find the object by values in it without having its identifier - use pkg/utils/object/compare helpers